src/Plugin/RulesDataMatcher/MixedLesserThanMatcher.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\RulesDataMatcher\MixedLesserThanMatcher.
+ */
+
+namespace Drupal\rules\Plugin\RulesDataMatcher;
+
+use Drupal\rules\Context\DataMatcherInterface;
+
+/**
+ * Defines a 'lesser than' matcher.
+ *
+ * @RulesDataMatcher(
+ *   id = "rules_mixed_lesser_than",
+ *   label = @Translation("A 'lesser than' matcher."),
```

Comment:
This kind of labels won't work for the select when having a form like
subject MATCHER-LABEL object
Thus, we need labels like "contains" "equals" etc. . It also shows the problem with the equality matcher - I guess it would be easier to handle if it's only one that works with every data type. Strict matching doesn't make much sense as Rules does not enforce any PHP types anyway and could be removed - so a "float" variable can be a PHP integer. -> Let's make an "any" "any" equals matcher - not sure how to add in the case sensitivity option best, but we can figure that out when we build the UI.